js/main.js

```diff
@@ -0,0 +1,144 @@
+
+const canvas = document.getElementById('wordSearchCanvas');
+const context = canvas.getContext('2d');
+
+
+function redraw() {
+  let height = canvas.getBoundingClientRect().height;
+  let width = canvas.getBoundingClientRect().width;
+
+  // console.log('Height is: ' + height + ' Width is: ' + width);
+}
+
+/**
+ * Returns a word seach game board that is created based of the dimensions
+ * of the allocated screen space with each letter being allocated it's
+ * own square within the board.
+ * @param {number} canvasHeight
+ * @param {number} canvasWidth
+ * @returns {Object}
```

Comment:
Define a record for all object-like types. For example for GameBoard it can be:

```js
/** @record */
class GameBoard {
  constructor() {
    /** @type {number} */
    this.startPointX;
    /** @type {number} */
    this.startPointY;
    ...
    /** @type {!Array<!Array<!Square>> */
    this.board;
  }
}
```

Then you can do

```js
const /** !GameBoard */ gameBoard = {startPointX: 0, ... }
```

and compiler will check that types and fields match correctly.

Reply:
Changed.